internal/blocksync/reactor.go

```diff
@@ -313,6 +326,39 @@ func (r *Reactor) processBlockSyncCh(ctx context.Context, blockSyncCh *p2p.Chann
 	}
 }
 
+// autoRestartIfBehind will check if the node is behind the max peer height by
+// a certain threshold. If it is, the node will attempt to restart itself
+func (r *Reactor) autoRestartIfBehind(ctx context.Context) {
```

Comment:
It seems we do not check whether the node is in blocksync mode already or not? What if the node is already in blocksync mode doing catch up, in that case we should not restart until it caught up right?

Reply:
Good point, let me add a check to see if it's already in blockSync